### DISCUSSION TITLE
error TS2411: Property 'y' of type 'ScaleLimits

### DISCUSSION THREAD

**Discussion by DarkvoiceGit** [chartjs/chartjs-plugin-zoom#565] [Q&A] [answered]
Hi I wanted to implement the zoom pan plugin into my chart, so I followed the guide

-  npm install chartjs-plugin-zoom
- import zoomPlugin from 'chartjs-plugin-zoom';
- added zoomPlugin to  the Chart.register(...)
- added the zoom into the chart like this
```
              plugins:{
                tooltip:{
                  enabled:false,
                  external: externalTooltipHandler
                  
                  
                },
                zoom:{
                  limits: {
                    x: {min: -200, max: 200, minRange: 50},
                    y: {min: -200, max: 200, minRange: 50}
                  },
                  pan: {
                    enabled: true,
                    mode: 'xy',
                  },
                  zoom :{
                    wheel:{
                      enabled:true
                    },
                    pinch: {
                      enabled: true
                    },
                    mode:'x',
                  }
                }
              }
```
I get this error

> Error: node_modules/chartjs-plugin-zoom/types/index.d.ts:50:30 - error TS7006: Parameter 'chart' implicitly has an 'any' type.
> 50 export function getZoomLevel(chart): number;
>                                ~~~~~
> Error: node_modules/chartjs-plugin-zoom/types/options.d.ts:173:3 - error TS2411: Property 'x' of type 'ScaleLimits | undefined' is not assignable to string index type 'ScaleLimits'.
> 173   x?: ScaleLimits;
>      ~
>Error: node_modules/chartjs-plugin-zoom/types/options.d.ts:174:3 - error TS2411: Property 'y' of type 'ScaleLimits | undefined' is not assignable to string index type 'ScaleLimits'.
>174   y?: ScaleLimits;

**Comment by kurkle** [comment, marked as answer]
https://github.com/chartjs/chartjs-plugin-zoom/issues/562

Is fixed in master and will be in v1.1.1
